documentation/modules/appendix/pages/best-practices.adoc

```diff
@@ -12,56 +12,854 @@ SPDX-License-Identifier: MPL-2.0
+* <<no-redundant-suffix>>
+* <<meaningful-names>>
+* <<add-payload-names-for-properties>>
+* <<if-certain-characteristics-are-present>>
+
+In addition to that, there are also best practices for:
+
+* <<the-preferredname-attribute>>
+* <<the-description-attribute>>
+
+[[camel-case-for-acronyms]]
+=== Camel case for acronyms
+
+For every part of a `name`, also for acronyms, strictly apply camel-case capitalization as detailed in the  xref:ROOT:modeling-guidelines.adoc#naming-rules[naming rules].
+
+For example:
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`mainECU`
+|`mainEcu`
+|`plantID`
+|`plantId`
+|`documentURL`
+|`documentUrl`
+|===
+
+Create `name` attributes where each capital letter can be trusted to trigger the start of a new name part.
+
+That way, for example, automated case conversion will produce meaningful names (e.g., when generating code based on an Aspect Model):
+
+|===
+|SAMM `name` |Resulting name in generated code (automated case conversion)
+
+|{nok-small} `documentURL`
+|`DOCUMENT_U_R_L` {nok-small}
+|{ok-small} `documentUrl`
+|`DOCUMENT_URL` {ok-small}
+|===
+
+[[unmistakable-abbreviations]]
+=== Unmistakable abbreviations
+
+Avoid abbreviations that are not sufficiently common.
+Seemingly obvious abbreviations in one context can be misunderstood in different contexts.
+
+For example:
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`temp`
+|Abbreviation unclear; not recommended. (Could mean "temperature" or "temporary".)
+|Abbreviating “minimum” or “maximum” by `min` or `max` is common enough and, therefore, acceptable.
+|`minTemperature` +
+`maxSpeed`
+|===
+
+Using unmistakable abbreviations also applies to <<abbreviations-in-descriptions,Description texts>>.
+
+[[no-redundant-prefix]]
+=== No redundant prefix
+
+Do not duplicate information.
+
+Avoid prefixes that repeat a domain (the namespace) an Aspect Model belongs to.
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`fooCount` +
+`fooId` +
+`fooXyz` +
+etc. +
+&nbsp; +
+Complete identifier of such a Property: +
+`urn:samm:foo.myapplication:1.0.0#fooCount`
+|`count` +
+`id` +
+`xyz` +
+etc. +
+&nbsp; +
+Complete identifier of such a Property: +
+`urn:samm:foo.myapplication:1.0.0#count`
+|===
+
+NOTE: The complete identifier of a Property is the versioned namespace combined with the element's `name`, for example, `urn:samm:foo.myapplication:1.0.0#count`.
+So, the `foo` context (namespace) is already part of `count`&#8203;'s identifier.
+
+[[no-redundant-suffix]]
+=== No redundant suffix
+
+Do not duplicate information.
+
+Avoid suffixes that repeat the SAMM model element type, such as Aspect, Property, etc.
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`MovementAspect` +
+&nbsp; +
+Result in TTL file: +
+`:MovementAspect a samm:Aspect ;`
+|`Movement` +
+&nbsp; +
+Result in TTL file: +
+`:Movement a samm:Aspect ;`
+|===
+
+TIP: To distinguish between, for example, an Entity and a related Characteristic, it is acceptable to add corresponding suffixes.
```

Comment:
The TIP is probably: "If the names do not appear in the payload (see https://eclipse-esmf.github.io/samm-specification/snapshot/payloads.html) then adding suffix might be useful. This holds for Entities and Characteristics, especially if a characteristic is defined specifically for a single property. Then we would have "<property name>" + ">property name>Characteristic.